Test Failure: PersistentExecutorWithFailoverEnabledTest.testBlockRunningTaskThatCancelsSelfFE times out and retries

Test Failure: com.ibm.ws.concurrent.persistent.fat.PersistentExecutorWithFailoverEnabledTest.testBlockRunningTaskThatCancelsSelfFE

```
testBlockRunningTaskThatCancelsSelfFE:junit.framework.AssertionFailedError: 2021-02-13-14:08:58:714 [Unexpected FFDC reporting javax.persistence.PersistenceException was found (count = 3): /home/jazz_build/Build/jbe/build/dev/image/output/wlp/usr/servers/com.ibm.ws.concurrent.persistent.fat.failover/logs/ffdc/ffdc_21.02.13_14.08.49.1.log
>------Start of DE processing------ = [2/13/21 14:08:49:828 PST]
>Exception = javax.persistence.PersistenceException
>Source = com.ibm.ws.concurrent.persistent.internal.PersistentExecutorImpl
>probeid = 965
>Stack Dump = javax.persistence.PersistenceException: 
>Exception Description: Error binding to externally managed transaction
>Internal Exception: javax.transaction.RollbackException: Transaction rolled back
>	at org.eclipse.persistence.internal.jpa.transaction.JTATransactionWrapper.registerIfRequired(JTATransactionWrapper.java:145)
>	at org.eclipse.persistence.internal.jpa.EntityManagerImpl.setJTATransactionWrapper(EntityManagerImpl.java:2168)
>	at org.eclipse.persistence.internal.jpa.EntityManagerImpl.detectTransactionWrapper(EntityManagerImpl.java:893)
>	at org.eclipse.persistence.internal.jpa.EntityManagerImpl.initialize(EntityManagerImpl.java:416)
>	at org.eclipse.persistence.internal.jpa.EntityManagerImpl.<init>(EntityManagerImpl.java:409)
>	at org.eclipse.persistence.internal.jpa.EntityManagerFactoryDelegate.createEntityManagerImpl(EntityManagerFactoryDelegate.java:324)
>	at org.eclipse.persistence.internal.jpa.EntityManagerFactoryImpl.createEntityManagerImpl(EntityManagerFactoryImpl.java:337)
>	at org.eclipse.persistence.internal.jpa.EntityManagerFactoryImpl.createEntityManager(EntityManagerFactoryImpl.java:303)
>	at com.ibm.wsspi.persistence.internal.PersistenceServiceUnitImpl.createEntityManager(PersistenceServiceUnitImpl.java:104)
>	at com.ibm.ws.concurrent.persistent.db.DatabaseTaskStore.findById(DatabaseTaskStore.java:484)
>	at com.ibm.ws.concurrent.persistent.internal.PersistentExecutorImpl.getStatus(PersistentExecutorImpl.java:964)
>	at web.SelfCancelingTask.call(SelfCancelingTask.java:86)
>	at web.SelfCancelingTask.call(SelfCancelingTask.java:27)
>	at com.ibm.ws.concurrent.persistent.internal.InvokerTask.run(InvokerTask.java:447)
>	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:522)
>	at java.util.concurrent.FutureTask.run(FutureTask.java:277)
>	at com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:239)
>	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1160)
>	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
>	at java.lang.Thread.run(Thread.java:822)
>Caused by: javax.transaction.RollbackException: Transaction rolled back
>	at com.ibm.tx.jta.impl.TransactionImpl.registerSynchronization(TransactionImpl.java:2284)
>	at com.ibm.tx.jta.impl.TransactionImpl.registerSynchronization(TransactionImpl.java:2265)
>	at org.eclipse.persistence.internal.jpa.transaction.JTATransactionWrapper.registerIfRequired(JTATransactionWrapper.java:136)
]
	at componenttest.custom.junit.runner.FATRunner.blowup(FATRunner.java:300)
	at componenttest.custom.junit.runner.FATRunner.access$600(FATRunner.java:66)
	at componenttest.custom.junit.runner.FATRunner$1.evaluate(FATRunner.java:250)
	at componenttest.rules.repeater.RepeatTests$CompositeRepeatTestActionStatement.evaluate(RepeatTests.java:115)
	at componenttest.custom.junit.runner.FATRunner$2.evaluate(FATRunner.java:318)
	at componenttest.custom.junit.runner.FATRunner.run(FATRunner.java:171)
```

A persistent task that is scheduled by the test hits a transaction timeout and rolls back, leading to the FFDCs and error messages. It is retried and runs cleanly and the test logic even passes. Given that retry of failed tasks is the intended design, everything looks to be working well and the test just needs to be updated to tolerate the FFDC/error messages.

--- a/dev/com.ibm.ws.concurrent.persistent_fat/fat/src/com/ibm/ws/concurrent/persistent/fat/PersistentExecutorWithFailoverEnabledTest.java
+++ b/dev/com.ibm.ws.concurrent.persistent_fat/fat/src/com/ibm/ws/concurrent/persistent/fat/PersistentExecutorWithFailoverEnabledTest.java
@@ -47,125 +47,125 @@ import web.SchedulerFATServlet;
 @RunWith(FATRunner.class)
 public class PersistentExecutorWithFailoverEnabledTest extends FATServletClient {
 
-	private static final String APP_NAME = "schedtest";
+    private static final String APP_NAME = "schedtest";
 
-	@ClassRule
-	public static RepeatTests r = RepeatTests.withoutModification()//
-			.andWith(new JakartaEE9Action());
+    @ClassRule
+    public static RepeatTests r = RepeatTests.withoutModification()//
+            .andWith(new JakartaEE9Action());
 
-	@Server("com.ibm.ws.concurrent.persistent.fat.failover")
-	@TestServlet(servlet = SchedulerFATServlet.class, path = APP_NAME)
-	public static LibertyServer server;
-	
-	public static final JdbcDatabaseContainer<?> testContainer = DatabaseContainerFactory.create();
+    @Server("com.ibm.ws.concurrent.persistent.fat.failover")
+    @TestServlet(servlet = SchedulerFATServlet.class, path = APP_NAME)
+    public static LibertyServer server;
 
-	@BeforeClass
-	public static void setUp() throws Exception {
-		testContainer.start();
-		
-		server.addEnvVar("DB_DRIVER", DatabaseContainerType.valueOf(testContainer).getDriverName());
+    public static final JdbcDatabaseContainer<?> testContainer = DatabaseContainerFactory.create();
 
-		DatabaseContainerUtil.setupDataSourceProperties(server, testContainer);
+    @BeforeClass
+    public static void setUp() throws Exception {
+        testContainer.start();
 
-		ShrinkHelper.defaultDropinApp(server, APP_NAME, "web");
+        server.addEnvVar("DB_DRIVER", DatabaseContainerType.valueOf(testContainer).getDriverName());
 
-		// Use the Jakarta version of test features if Jakarta is being used.
-		if (JakartaEE9Action.isActive()) {
-			ServerConfiguration config = server.getServerConfiguration();
-			Set<String> features = config.getFeatureManager().getFeatures();
-			features.remove("persistentexecutor-1.0");
-			features.remove("timerinterfacestestfeature-1.0");
-			features.add("persistentExecutor-2.0");
-			features.add("timerInterfacesTestFeature-2.0");
-			server.updateServerConfiguration(config);
-		}
+        DatabaseContainerUtil.setupDataSourceProperties(server, testContainer);
 
-		server.startServer();
-	}
+        ShrinkHelper.defaultDropinApp(server, APP_NAME, "web");
 
-	@AfterClass
-	public static void tearDown() throws Exception {
-		try {
-			runTest(server, APP_NAME, "verifyNoTasksRunning");
-		} finally {
-			try {
-				if (server.isStarted())
-					server.stopServer("CWWKC1500W", // Task rolled back
-							"CWWKC1501W", // Task rolled back due to failure ...
-							"CWWKC1502W", // Task rolled back, retry time unspecified
-							"CWWKC1503W", // Task rolled back due to failure ..., retry time unspecified
-							"CWWKC1510W", // Task rolled back and aborted
-							"CWWKC1511W", // Task rolled back and aborted. Failure is ...
-							"DSRA0174W",  // Generic Datasource Helper
-							"DSRA030*E"); // XA errors related to transaction timeout
-			} finally {
-				testContainer.stop();
-			}
-		}
-	}
+        // Use the Jakarta version of test features if Jakarta is being used.
+        if (JakartaEE9Action.isActive()) {
+            ServerConfiguration config = server.getServerConfiguration();
+            Set<String> features = config.getFeatureManager().getFeatures();
+            features.remove("persistentexecutor-1.0");
+            features.remove("timerinterfacestestfeature-1.0");
+            features.add("persistentExecutor-2.0");
+            features.add("timerInterfacesTestFeature-2.0");
+            server.updateServerConfiguration(config);
+        }
 
-	@Test
-	public void testBlockAfterCancelByIdFE() throws Exception {
-		runTest(server, APP_NAME, "testBlockAfterCancelByIdFE");
-	}
+        server.startServer();
+    }
 
-	@Test
-	public void testBlockAfterCancelByNameFE() throws Exception {
-		runTest(server, APP_NAME, "testBlockAfterCancelByNameFE");
-	}
+    @AfterClass
+    public static void tearDown() throws Exception {
+        try {
+            runTest(server, APP_NAME, "verifyNoTasksRunning");
+        } finally {
+            try {
+                if (server.isStarted())
+                    server.stopServer("CWWKC1500W", // Task rolled back
+                            "CWWKC1501W", // Task rolled back due to failure ...
+                            "CWWKC1502W", // Task rolled back, retry time unspecified
+                            "CWWKC1503W", // Task rolled back due to failure ..., retry time unspecified
+                            "CWWKC1510W", // Task rolled back and aborted
+                            "CWWKC1511W", // Task rolled back and aborted. Failure is ...
+                            "DSRA0174W",  // Generic Datasource Helper
+                            "DSRA030*E"); // XA errors related to transaction timeout
+            } finally {
+                testContainer.stop();
+            }
+        }
+    }
 
-	@Test
-	public void testBlockAfterFindByIdFE() throws Exception {
-		runTest(server, APP_NAME, "testBlockAfterFindByIdFE");
-	}
+    @Test
+    public void testBlockAfterCancelByIdFE() throws Exception {
+        runTest(server, APP_NAME, "testBlockAfterCancelByIdFE");
+    }
 
-	@Test
-	public void testBlockAfterFindByNameFE() throws Exception {
-		runTest(server, APP_NAME, "testBlockAfterFindByNameFE");
-	}
+    @Test
+    public void testBlockAfterCancelByNameFE() throws Exception {
+        runTest(server, APP_NAME, "testBlockAfterCancelByNameFE");
+    }
 
-	@Test
-	public void testBlockAfterRemoveByIdFE() throws Exception {
-		runTest(server, APP_NAME, "testBlockAfterRemoveByIdFE");
-	}
+    @Test
+    public void testBlockAfterFindByIdFE() throws Exception {
+        runTest(server, APP_NAME, "testBlockAfterFindByIdFE");
+    }
 
-	@Test
-	public void testBlockAfterRemoveByNameFE() throws Exception {
-		runTest(server, APP_NAME, "testBlockAfterRemoveByNameFE");
-	}
+    @Test
+    public void testBlockAfterFindByNameFE() throws Exception {
+        runTest(server, APP_NAME, "testBlockAfterFindByNameFE");
+    }
 
-	@Test
-	public void testBlockAfterScheduleFE() throws Exception {
-		runTest(server, APP_NAME, "testBlockAfterScheduleFE");
-	}
+    @Test
+    public void testBlockAfterRemoveByIdFE() throws Exception {
+        runTest(server, APP_NAME, "testBlockAfterRemoveByIdFE");
+    }
 
-	@Test
-	public void testBlockRunningTaskFE() throws Exception {
-		runTest(server, APP_NAME, "testBlockRunningTaskFE");
-	}
+    @Test
+    public void testBlockAfterRemoveByNameFE() throws Exception {
+        runTest(server, APP_NAME, "testBlockAfterRemoveByNameFE");
+    }
 
-	@Test
-	public void testBlockRunningTaskThatCancelsSelfFE() throws Exception {
-		runTest(server, APP_NAME, "testBlockRunningTaskThatCancelsSelfFE");
-	}
+    @Test
+    public void testBlockAfterScheduleFE() throws Exception {
+        runTest(server, APP_NAME, "testBlockAfterScheduleFE");
+    }
 
-	@Test
-	public void testBlockRunningTaskThatRemovesSelfFE() throws Exception {
-		runTest(server, APP_NAME, "testBlockRunningTaskThatRemovesSelfFE");
-	}
+    @Test
+    public void testBlockRunningTaskFE() throws Exception {
+        runTest(server, APP_NAME, "testBlockRunningTaskFE");
+    }
 
-	@Test
-	public void testCancelRunningTaskFE() throws Exception {
-		runTest(server, APP_NAME, "testCancelRunningTaskFE");
-	}
+    @Test
+    public void testBlockRunningTaskThatCancelsSelfFE() throws Exception {
+        runTest(server, APP_NAME, "testBlockRunningTaskThatCancelsSelfFE");
+    }
 
-	@Test
-	public void testRemoveRunningTaskAutoPurgeFE() throws Exception {
-		runTest(server, APP_NAME, "testRemoveRunningTaskAutoPurgeFE");
-	}
+    @Test
+    public void testBlockRunningTaskThatRemovesSelfFE() throws Exception {
+        runTest(server, APP_NAME, "testBlockRunningTaskThatRemovesSelfFE");
+    }
 
-	@Test
-	public void testRemoveRunningTaskFE() throws Exception {
-		runTest(server, APP_NAME, "testRemoveRunningTaskFE");
-	}
+    @Test
+    public void testCancelRunningTaskFE() throws Exception {
+        runTest(server, APP_NAME, "testCancelRunningTaskFE");
+    }
+
+    @Test
+    public void testRemoveRunningTaskAutoPurgeFE() throws Exception {
+        runTest(server, APP_NAME, "testRemoveRunningTaskAutoPurgeFE");
+    }
+
+    @Test
+    public void testRemoveRunningTaskFE() throws Exception {
+        runTest(server, APP_NAME, "testRemoveRunningTaskFE");
+    }
 }

--- a/dev/com.ibm.ws.concurrent.persistent_fat/fat/src/com/ibm/ws/concurrent/persistent/fat/PersistentExecutorWithFailoverEnabledTest.java
+++ b/dev/com.ibm.ws.concurrent.persistent_fat/fat/src/com/ibm/ws/concurrent/persistent/fat/PersistentExecutorWithFailoverEnabledTest.java
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2014, 2020 IBM Corporation and others.
+ * Copyright (c) 2014, 2021 IBM Corporation and others.
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License v1.0
  * which accompanies this distribution, and is available at
@@ -35,9 +35,14 @@ import componenttest.topology.impl.LibertyServer;
 import componenttest.topology.utils.FATServletClient;
 import web.SchedulerFATServlet;
 
-@AllowedFFDC({ "javax.resource.ResourceException", // due to transaction timeout from infra slowness
-		"javax.transaction.RollbackException", // due to transaction timeout from infra slowness
-		"javax.transaction.xa.XAException" // due to transaction timeout from infra slowness
+@AllowedFFDC({
+    "jakarta.persistence.PersistenceException", // after transaction times out and rolls back
+    "jakarta.resource.ResourceException", // due to transaction timeout from infra slowness
+    "jakarta.transaction.RollbackException", // due to transaction timeout from infra slowness
+    "javax.persistence.PersistenceException", // after transaction times out and rolls back
+    "javax.resource.ResourceException", // due to transaction timeout from infra slowness
+    "javax.transaction.RollbackException", // due to transaction timeout from infra slowness
+    "javax.transaction.xa.XAException" // due to transaction timeout from infra slowness
 })
 @RunWith(FATRunner.class)
 public class PersistentExecutorWithFailoverEnabledTest extends FATServletClient {
@@ -91,7 +96,8 @@ public class PersistentExecutorWithFailoverEnabledTest extends FATServletClient 
 							"CWWKC1503W", // Task rolled back due to failure ..., retry time unspecified
 							"CWWKC1510W", // Task rolled back and aborted
 							"CWWKC1511W", // Task rolled back and aborted. Failure is ...
-							"DSRA0174W"); // Generic Datasource Helper
+							"DSRA0174W",  // Generic Datasource Helper
+							"DSRA030*E"); // XA errors related to transaction timeout
 			} finally {
 				testContainer.stop();
 			}

